public/pages/CreateMonitor/containers/CreateMonitor/utils/helpers.js

```diff
@@ -0,0 +1,255 @@
+
+export const getInitialValues = ({
+  title,
+  index,
+  timeField,
+  flyoutMode,
+  location,
+  monitorToEdit,
+  edit,
+  searchType,
+  detectorId,
+  embeddable,
+}) => {
+  let initialValues = _.mergeWith(
+    {},
+    _.cloneDeep(FORMIK_INITIAL_VALUES),
+    initializeFromQueryParams(queryString.parse(location.search)),
+    (initialValue, queryValue) => (_.isEmpty(queryValue) ? initialValue : queryValue)
+  );
+
+  if (flyoutMode) {
+    initialValues.name = `${title} ${getDigitId()}`;
+    initialValues.index = index;
+    initialValues.timeField = timeField;
+
+    // Add trigger
+    const monitorType = initialValues.monitor_type;
+    const initialTrigger = getInitialTriggerValues({ flyoutMode, monitorType, triggers: [] });
+    initialValues.triggerDefinitions = [initialTrigger];
+
+    // Add aggregations
+    initialValues.aggregations = getMetricAgg(embeddable);
+
+    if (searchType) {
+      initialValues.searchType = searchType;
+    }
+
+    if (detectorId) {
+      initialValues.detectorId = detectorId;
+      initialValues.period = { interval: 20, unit: 'MINUTES' };
+    }
+  }
+
+  if (edit && monitorToEdit) {
+    const triggers = triggerToFormik(_.get(monitorToEdit, 'triggers', []), monitorToEdit);
+    initialValues = {
+      ...monitorToFormik(monitorToEdit),
+      triggerDefinitions: triggers.triggerDefinitions,
+    };
+  }
+
+  return initialValues;
+};
+
+const getMetricAgg = (embeddable) => {
+  let aggregationType = AGGREGATION_TYPES[1].value;
+  let fieldName = '';
+  if (embeddable?.vis?.data?.aggs?.aggs.length === 1) {
+    const agg = embeddable.vis.data.aggs.aggs[0];
+    if (agg.schema === 'metric' && !(aggregationType && fieldName) && agg.params.field) {
+      console.log(agg);
+      aggregationType = agg.__type.dslName;
+      fieldName = agg.params.field.spec.name;
+    }
+  }
+  return [
+    {
+      aggregationType: aggregationType,
+      fieldName: fieldName,
+    },
+  ];
+};
+
+export const getPlugins = async (httpClient) => {
+  try {
+    const pluginsResponse = await httpClient.get('../api/alerting/_plugins');
+    if (pluginsResponse.ok) {
+      return pluginsResponse.resp.map((plugin) => plugin.component);
+    } else {
+      console.error('There was a problem getting plugins list');
+      return [];
+    }
+  } catch (e) {
+    console.error('There was a problem getting plugins list', e);
+    return [];
+  }
+};
+
+export const prepareTriggers = ({
+  trigger,
+  triggerMetadata,
+  monitor,
+  edit,
+  triggerToEdit = [],
+}) => {
+  const { ui_metadata: uiMetadata = {}, triggers, monitor_type } = monitor;
+  let updatedTriggers;
+  let updatedUiMetadata;
+
+  if (edit) {
+    updatedTriggers = _.isArray(trigger) ? trigger.concat(triggers) : [trigger].concat(triggers);
+    updatedUiMetadata = {
+      ...uiMetadata,
+      triggers: { ...uiMetadata.triggers, ...triggerMetadata },
+    };
+  } else {
+    let updatedTriggersMetadata = _.cloneDeep(uiMetadata.triggers || {});
```

Comment:
Make this a `const`

Reply:
Should be fine even when we want to delete an attribute from it?